src/precice/impl/Participant.cpp

```diff
@@ -103,15 +103,17 @@ void Participant::addWriteData(
     const mesh::PtrMesh &mesh)
 {
   checkDuplicatedData(data);
-  _writeDataContexts[data->getID()] = std::make_unique<WriteDataContext>(data, mesh);
+  auto inserted = _writeDataContexts.insert(std::make_pair(data->getID(), WriteDataContext(data, mesh)));
+  PRECICE_ASSERT(inserted.second, "addWriteData failed.");
 }
 
 void Participant::addReadData(
     const mesh::PtrData &data,
     const mesh::PtrMesh &mesh)
 {
   checkDuplicatedData(data);
-  _readDataContexts[data->getID()] = std::make_unique<ReadDataContext>(data, mesh);
+  auto inserted = _readDataContexts.insert(std::make_pair(data->getID(), ReadDataContext(data, mesh)));
+  PRECICE_ASSERT(inserted.second, "addReadData failed.");
```

Comment:
same as above

Reply:
Removed assertion and using `emplace` in  69c6da5b425916ddfdb8faa33c93fbedc5f33d21.